lm_eval/tasks/diabla.py

```diff
@@ -0,0 +1,85 @@
+# TODO: Remove all TODO comments once the implementation is complete.
+"""
+DiaBLa: English-French Bilingual dialogue dataset for Machine Translation
+https://link.springer.com/article/10.1007/s10579-020-09514-4
+
+Rachel Bawden, Eric Bilinski, Thomas Lavergne and Sophie Rosset
+(2021). DiaBLa: A Corpus of Bilingual Spontaneous Written Dialogues
+for Machine Translation. Language Resources and Evaluation(55). Pages
+635–660. Springer Verlag. 10.1007/s10579-020-09514-4.
+
+DiaBLa is an English-French dataset for the evaluation of Machine
+Translation (MT) for informal, written bilingual dialogue.  It
+contains 144 spontaneous dialogues (5,700+ sentences) between native
+English and French speakers, mediated by one of two neural MT systems
+in a range of role-play settings. The dialogues are accompanied by
+fine-grained sentence-level judgments of MT quality, produced by the
+dialogue participants themselves, as well as by manually normalised
+versions and reference translations produced a posteriori
+
+Homepage: http://almanach.inria.fr/software_and_resources/custom/DiaBLa-en.html
+"""
+from lm_eval.base import PromptSourceTask
+from typing import List
+
+_CITATION = """@article{bawden_DiaBLa:-A-Corpus-of_2021,
+  author = {Bawden, Rachel and Bilinski, Eric and Lavergne, Thomas and Rosset, Sophie},
+  doi = {10.1007/s10579-020-09514-4},
+  title = {DiaBLa: A Corpus of Bilingual Spontaneous Written Dialogues for Machine Translation},
+  year = {2021},
+  journal = {Language Resources and Evaluation},
+  publisher = {Springer Verlag},
+  volume = {55},
+  pages = {635--660},
+  url = {https://hal.inria.fr/hal-03021633},
+  pdf = {https://hal.inria.fr/hal-03021633/file/diabla-lre-personal-formatting.pdf},
+}
+"""
+
+
+class DiaBLa(PromptSourceTask):
+    VERSION = 0
+    DATASET_PATH = "rbawden/DiaBLa"
+    DATASET_NAME = None
+
+    def has_training_docs(self):
+        return False
+
+    def has_validation_docs(self):
+        return False
+
+    def has_test_docs(self):
+        return True
+
+    def training_docs(self):
+        if self.has_training_docs():
+            if self._training_docs is None:
+                self._training_docs = list(self.dataset["train"])
+            return self._training_docs
+
+    def validation_docs(self):
+        if self.has_validation_docs():
+            return self.dataset["validation"]
+
+    def test_docs(self):
+        if self.has_test_docs():
+            return self.dataset["test"]
+
+    def max_generation_length(self):
+        return 512
+
+    def stopping_criteria(self):
+        return "\n"
```

Comment:
Note that by doing this you override the default stopping criterion for few-shot prompting, namely, `"\n###\n"`. 
Also, `t5` based tokenizers will ignore newlines rendering this stopping criterion useless for them. 

Is relying on `max_generation_length` alone not enough to obtain reasonable generations in the 0-shot setting?

Reply:
I did some initial tests (with gpt2, even though it's not adapted to multilingual - just to test what happened) and it seemed to keep generating really long sequences (including repeating the same good sequence over and over again, using \n to separate them). I can certainly try to change this - it needs to be compatible with all models, but I wasn't sure what do about this. `max_generation_length` is tricky because you can't tell in advance how long a sequence is going to be and we are relying on the model being good enough to only generate the necessary tokens and not up to the maximum length. I wanted to run some more tests, but I've run into some difficulties. When I run the evaluation (just testing on a few examples), I now get `AttributeError: 'NoneType' object has no attribute 'apply'`. I've done a `pip install` with my local version of promptsource, so it should be able to see the prompt templates. So unfortunately I can't test whether the default stopping criterion is ok. Do you have any ideas how I can get round this error? I have uninstalled and reinstalled several times. Thanks in advance for your help!

Reply:
I ended up finding the solution for the install issues - I forked promptsource many months ago and therefore didn't have the eval-hackathon branch, so it was saying it was up-to-date when it wasn't, so I managed to figure it out :-) 

I have pushed a latest version without the stopping criterion, but I don't know whether it is worth accepting the PR straight away (and modifying it later) or waiting to define a more adequate stopping criterion. That depends on when tasks will be run - it would be good to get some baseline results, but if that means rerunning them again soon because there could have been a better stopping criterion used, that would be a shame. What do you think?

Ideally (and summarising the slack conversation), I think that at least for this dataset, the following criteria would be good:

- <eos> or \n depending on which comes first (since I don't expect to see \n in any of the targets) - so this would involve having multiple possible criteria (for zero-shot)
- last resort: maximum length (this would be ok, but I wouldn't want it to interfere with any otherwise good translations. It can be quite hard to predict the maximum possible length.

What are your thoughts on this and potentially waiting before accepting? We could also discuss it at the meeting later if you're going to be there. I imagine it affects a couple of the tasks, including all the MT ones.

Reply:
Ok, given the previous discussion, I think this is ok without defining any stopping criteria (at least for now ;-)). Are the changes ok?

Reply:
At least for now 😄 (I'll make a note to come back to this in case we run beam search somewhere down the road) Thanks, Rachel!